libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1615,15 +1615,27 @@ void ExpressionCompiler::appendExternalFunctionCall(
 		m_context.experimentalFeatureActive(ExperimentalFeature::V050) &&
 		m_context.evmVersion().hasStaticCall();
 
+	bool haveReturndatacopy = m_context.evmVersion().supportsReturndata();
 	unsigned retSize = 0;
+	TypePointers returnTypes;
 	if (returnSuccessCondition)
 		retSize = 0; // return value actually is success condition
+	else if (haveReturndatacopy)
+		returnTypes = _functionType.returnParameterTypes();
 	else
-		for (auto const& retType: _functionType.returnParameterTypes())
+		returnTypes = _functionType.returnParameterTypesWithoutDynamicTypes();
+
+	bool dynamicReturnSize = false;
+	for (auto const& retType: returnTypes)
+		if (retType->isDynamicallyEncoded())
```

Comment:
Don't we have `calldataEncodedSize` and `isDynamicallyEncoded` on the `TupleType`, do we need to iterate?

Reply:
Apaprently no, but would it make sense to do so?

Reply:
Tuples are not fully-fledged types in Solidity, so I'm not sure which implications this would have.

Reply:
Let's postpone it. Created issue #3769.